### PR TITLE
Moe Sync

### LINKED ISSUE #1287: InconsistentHashCode fails on JDK13 EA
Got this error while using the error prone gradle plugin with the most recent JDK13 EA build.   It says to report it here, so here I am.   

Repro:

1.  `git clone http://github.com/perfmark/perfmark`
2.  `cd perfmark`
3.  `JAVA_HOME=~/jdk-13/ ./gradlew :perfmark-java9:jmh`

```
/home/carl/IdeaProjects/perfmark/impl-api/src/main/java/io/perfmark/impl/Mark.java:6: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
public final class Mark {
             ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.3.3
     BugPattern: InconsistentHashCode
     Stack Trace:
     java.lang.NullPointerException
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.handleSymbol(InconsistentHashCode.java:218)
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.visitIdentifier(InconsistentHashCode.java:213)
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.visitIdentifier(InconsistentHashCode.java:149)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCIdent.accept(JCTree.java:2331)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitBreak(TreeScanner.java:472)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBreak.accept(JCTree.java:1577)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:91)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:106)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:114)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitCase(TreeScanner.java:372)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCase.accept(JCTree.java:1293)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:91)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:106)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:114)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitSwitch(TreeScanner.java:333)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCSwitch.accept(JCTree.java:1235)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:106)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:249)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1038)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitLabeledStatement(TreeScanner.java:320)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCLabeledStatement.accept(JCTree.java:1206)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:91)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:106)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:249)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1038)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:87)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:91)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitMethod(TreeScanner.java:207)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:904)
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.scanMethod(InconsistentHashCode.java:172)
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.scanMethod(InconsistentHashCode.java:161)
        at com.google.errorprone.bugpatterns.InconsistentHashCode$FieldAccessFinder.access$000(InconsistentHashCode.java:149)
        at com.google.errorprone.bugpatterns.InconsistentHashCode.matchClass(InconsistentHashCode.java:117)
        at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:433)
        at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:534)
        at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:150)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:814)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at com.google.errorprone.scanner.Scanner.scan(Scanner.java:71)
        at com.google.errorprone.scanner.Scanner.scan(Scanner.java:45)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:106)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:114)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:145)
        at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:546)
        at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:150)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:597)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
        at com.google.errorprone.scanner.Scanner.scan(Scanner.java:55)
        at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:43)
        at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:151)
        at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1423)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1370)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:959)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
        at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
        at org.gradle.api.internal.tasks.compile.AnnotationProcessingCompileTask.call(AnnotationProcessingCompileTask.java:89)
        at org.gradle.api.internal.tasks.compile.ResourceCleaningCompilationTask.call(ResourceCleaningCompilationTask.java:57)
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:50)
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:36)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:100)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:52)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:38)
        at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:49)
        at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:35)
        at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:39)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilerFactory$2.execute(IncrementalCompilerFactory.java:110)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilerFactory$2.execute(IncrementalCompilerFactory.java:106)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:59)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:43)
        at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:153)
        at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:121)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:277)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:262)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.base/java.lang.Thread.run(Thread.java:830)
1 error
3 warnings

FAILURE: Build failed with an exception.

```

### PR DESCRIPTION
**Pull request by ronshapiro**
This code has been reviewed and submitted internally. Feel free to discuss on the PR and we can submit follow-up changes as necessary.

Commits:
=====
<p> Replace google.github.io/dagger with dagger.dev

bb00ae194a3d126033eaa3c5d68f16707755d2b2

-------

<p> Flag instances of Duration.of(long, ChronoUnit) where the ChronoUnit is not an exact duration.

#badtime #goodtime

RELNOTES: Flag instances of Duration.of(long, ChronoUnit) where the ChronoUnit is not an exact duration.

221fde7d65d5cdd6ed0bd5a9a1fa7e5f4ab48acd

-------

<p> Fix an incorrect UnnecessaryAnonymousClass refactoring

don't refactor anonymous classes that implement the FI via a superclass.

0faf698a1563222527195638befec3429a4cdf80

-------

<p> Fix NPE on JDK13+ when InconsistentHashCode visits a break statement with a label.

Fixes #1287

RELNOTES: n/a

7bb2a776f215389d23637fe087e9a93ebba7ebe4

-------

<p> Expand SelfAssignment to cover the Precondition-style APIs in c.g.c.time.Durations,c.g.protobuf.util.{Durations,Timestamps}.

5ddc53ff5138ef6777855d1ea51d73b049507340

-------

<p> Allow ChronoUnit.DAYS explicitly in Duration.of(long, TemporalUnit).

(It's explicitly called out in the javadocs, but I missed it the first read.)

Also turn off the checker for desugar'ed java.time.

254f3f29e96da06e2aad0cb5851c5b7c2a237137

-------

<p> Rename DurationOfLongTemporalUnit to DurationTemporalUnit and cover duration.plus(long, TemporalUnit) and duration.minus(long, TemporalUnit).

Also ban calls to Instant APIs where the TemporalUnit is not supported.

beef3d89bc77ba88daddd867f87e154dbebe0dfc

-------

<p> Provide an example of using proto3's primitive wrappers.

e2cb9fed7eced3226fb6131715e4dbfafe3ef992

-------

<p> Migrate org.mockito.Matchers#any* to org.mockito.ArgumentMatchers

The former is deprecated and replaced by the latter in Mockito 2. However, there is a
functional difference: ArgumentMatchers will reject `null` and check the type
if the matcher specified a type (e.g. `any(Class)` or `anyInt()`). `any()` will
remain to accept anything.

c8750770a2384b42a3e9758a1ab2f29f7552e88f

-------

<p> Optimize VisitorState.inferBinaryName() by iterating directly over the chars of the class name.

This saves the need to create temporary strings for each piece of a package/each simple name of a type, uses StringBuilder.append(char) instead of append(String), and avoids some of the Iterable/Iterator/CharMatcher overhead. It also lends itself to easily checking if any replacement is necessary, which is one of the biggest savings this change brings.

RELNOTES: Performance improvements

8e131a229cc7a68fde8015115166951a24819574

-------

<p> Replace google.github.io/dagger with dagger.dev

fc362004ddf6bed5a92e1e3468c0f1420b33a133

-------

<p> Cache references to "JavacSingletonType.instance(context)" references within ErrorProne. Expose them as methods on VisitorState and only request them once

On their own, these seem insignificant, but in aggregate they're timing is noticeable.

266ffcc1f3333b50c91e89ff2304497ac7c0ebeb

-------

<p> Update docs for the Truth migration from an `actual()` method to an `actual` field.

9dc8eaf69e2eb743a92b924a3d615c13aee2bfca

-------

<p> Use the latest checker framework dataflow version

567b765bd7f36775faf0370a47049cd3ebb4fe65

-------

<p> Remove the custom link from UnusedException.

This already had documentation, which I guess wasn't being linked to. Winning.

ced971d6feb85ecfbd75f22a14f16372b3767979

-------

<p> Add valueOf to list of BadImport static imports

https://github.com/google/error-prone/issues/1284

21599b72de2fd98334d7a3bad9799fb9a14d8e39

-------

<p> Make LocalDate.plus/minus(Duration) a compile-time error.
#badtime #goodtime

d013d6a77aa37f51854c925ac3516f4e99c5d8c1

-------

<p> Add Class to list of bad imports.

We have nested protobuf classes that mask java.lang.Class.

64bf016c5265d26d05b0c943644444d0bad499cc

-------

<p> Update to Truth 0.45, and address deprecations.

Renames may include:
- containsAllOf => containsAtLeast
- containsAllIn => containsAtLeastElementsIn
- isSameAs => isSameInstanceAs
- isOrdered => isInOrder
- isStrictlyOrdered => isInStrictOrder

The other major change is to change custom subjects to extend raw Subject instead of supplying type parameters. The type parameters are being removed from Subject. This CL will temporarily produce rawtypes warnings, which will go away when I remove the type parameters (as soon as this batch of CLs is submitted).

Some CLs in this batch also migrate calls away from actualAsString(). Its literal replacement is `"<" + actual + ">"` (unless an object overrides actualCustomStringRepresentation()), but usually I've made a larger change, such as switching from an old-style "Not true that..." failure message to one generated with the Fact API. In that case, the new code usually contains a direct reference to this.actual (a field that I occasionally had to create). Another larger change I sometimes made is to switch from a manual check-and-fail approach to instead use check(...). And sometimes I just remove a withMessage() call that's no longer necessary now that the code uses check(...), or I introduce a check(...) call. (An assertion made with check(...) automatically includes the actual value from the original subject, so there's no need to set it again with withMessage().)

Finally, there's one CL in this batch in which I migrate a Correspondence subclass to instead use Correspondence.from.

1a8c86817aace47bbf8f25fe4d555f19e9c66565

-------

<p> Remove obsolete todo

3cdc1fd396f3be6e6b7435a99a08641b02afaaaf

-------

<p> Recommend Period in the LocalDateTemporalAmount summary.

d1aad72a63955bab2665755a9edb2a643bbf324d

-------

<p> Replace some for loops with stream operatoins

It reads more nicely without the builders, and I don't think it's in some performance-sensitive area where we should worry about the slight cost of a stream.

eae7d4dce89a2b2b3344712ead702799fa0a1fcd

-------

<p> Cleanups in CheckReturnValue

- Anonymous class to lambda
- Guava Optional to j.u.Optional
  - Replace ifPresent/get pairs with orElseGet chain
- Fix spelling mistakes in comments
- Remove an UnnecessaryLambda(TM)
- Remove some unused parameters

a7165448774abd68614a849d7b9ef88a9c81ad25

-------

<p> Move @CheckReturnValue to the class level

c0e3539b9ca9b10eef81307196274a24b411e95a

-------

<p> Clean up EmptyIfStatement

only get the parent once, remove an unnecessary assert.

205ad1a00087c96257d52dfe72fa436443959112

-------

<p> Don't repeat a bunch of work in MethodMatcher usage

bd513dd2e95e902d844410597cbbffd6e4a2bc19

-------

<p> Remove an unused parameter

b530fcabf234145fa427b81b05ea6e6e6bc2c333

-------

<p> Add the TemporalAmount implementations from org.threeten.extra to the list of well known immutable types.

All of them clearly document immutability in their contract, e.g.: https://www.threeten.org/threeten-extra/apidocs/org.threeten.extra/org/threeten/extra/Days.html

RELNOTES: N/A

fc972dbffb7d9da85849e4d009326247362ec977

-------

<p> Fix open-source breakages caused by my recent update to Truth 0.45:

- Update ImplementAssertionUsingChaining to operate on the new failWithActual and failWithoutActual methods instead of hardcoding the legacy methods. (The legacy methods have been removed externally.)

- Update TruthSelfEquals to check isSameInstanceAs instead of isSameAs. (isSameAs has been removed externally.)

- Update UseCorrectAssertInTests to generate isSameInstanceAs instead of isSameAs. (isSameAs has been removed externally.)

- Upgrade Compile-Testing to 0.18. (0.16 isn't compatible with the new Truth.)

a63e46f7d23938e29003434dcd89e5b94f41a986

-------

<p> Make Matchers.parentNode type-safer

47dfcb41b63b326c5916ef5988a96f064147322d

-------

<p> Add a TODO to revisit nextStatement's handling of the last statement in a block

e3766e4d42d2b9cfed92f2eb70b90e8a32b345b1

-------

<p> Make methodName a String

contentEquals() will do that anyway, so this saves us 3 String creations.

Alternatively, we could use Names to cache the names and then use equals() on those. I'm open to that if we'd like.

163a2b74276808191fa07ca3d3101cdcfda00989

-------

<p> Discourage calling #toString on lite protos.

Ignore cases which are clearly being called within a debug/verbose log method, or in testonly code.

4ed49efb44eb5124a0c08a6d354966153334f97c

-------

<p> Add a check for calling checkNotNull twice on the same variable.

0fbefb0573e0d1ad9a0f7976e4f953adc4162587

-------

<p> Migrate AbstractToString onto java.util.Optional

705247495bf36cce5f5f45832b618331146a25ac

-------

<p> Fix typo in UnsafeFinalization.md

231d8b6ccde04b5d843966b7432b9a246a1d4d4e

-------

<p> Turn off ErrorProne for large code generators.

Currently only protoc and Dagger-generated files are supported. Security checks (@BugPattern(disableable = false)) are still invoked to make sure the generators don't generate any dangerous code.

RELNOTES: N/A

03f19c9ccc1ba559c87f39edf6c6853471b6146e

### CODE REVIEW

File: check_api/src/main/java/com/google/errorprone/ErrorProneOptions.java [hunk 1]
```diff
@@ -221,6 +225,10 @@ public boolean isIgnoreSuppressionAnnotations() {
     return ignoreSuppressionAnnotations;
   }
 
+  public boolean ignoreLargeCodeGenerators() {
```
Comment (Stephan202) [inline check_api/src/main/java/com/google/errorprone/ErrorProneOptions.java]: While this flag is exposed, it seems the associated filtering logic hasn't been open-sourced. Are there plans to do so? This looks like an interesting alternative to `-XepDisableWarningsInGeneratedCode` (if I understand correctly what it does).

File: core/src/main/java/com/google/errorprone/bugpatterns/time/DurationTemporalUnit.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+
+/**
+ * Bans calls to {@code Duration} APIs where the {@link java.time.temporal.TemporalUnit} is not
+ * {@link ChronoUnit#DAYS} or it has an estimated duration (which is guaranteed to throw an {@code
```
Comment (Stephan202) [inline core/src/main/java/com/google/errorprone/bugpatterns/time/DurationTemporalUnit.java]:
```suggestion
 * {@link ChronoUnit#DAYS} and has an estimated duration (which is guaranteed to throw an {@code
```
Repeated this comment just so it doesn't get lost following closure of #1291.

File: docs/bugpattern/ChainedAssertionLosesContext.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ Before:
 ```
 class MyProtoSubject {
   public void hasFoo(Foo expected) {
-    assertThat(actual().foo()).isEqualTo(expected);
+    assertThat(actual.foo()).isEqualTo(expected);
```
Comment (jbduncan) [inline docs/bugpattern/ChainedAssertionLosesContext.md]: AFAICT, `MyProtoSubject` lacks an `actual` variable or field, so I wonder if it should be updated? :thinking:

File: docs/bugpattern/ChainedAssertionLosesContext.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ After:
 ```
 class MyProtoSubject {
   public void hasFoo(Foo expected) {
-    check("foo()").that(actual().foo()).isEqualTo(expected);
+    check("foo()").that(actual.foo()).isEqualTo(expected);
```
Comment (jbduncan) [inline docs/bugpattern/ChainedAssertionLosesContext.md]: This example lacks an `actual` variable or field, so I wonder if it should be updated?

**Comment by ronshapiro** [comment]
@stephan202 sorry for resolving your comment, I meant to reply. So here's my response:

We are currently evaluating that flag in an experiment. It's possible that we'll decide to opensource that logic after we have a better handle on it and if it's the right approach to what we want to be doing.

The reason that some of the methods were opensourced is primarily because it was expansive enough that it was a lot to try to strip it all, so I figured that it stripping the ability to set it was sufficient for the time being

**Comment by Stephan202** [comment]
@ronshapiro clear, thanks!

File: docs/bugpattern/ChainedAssertionLosesContext.md (hunk 3 above)
Comment (cpovirk) [reply docs/bugpattern/ChainedAssertionLosesContext.md]: The class is missing a constructor and lots of other things, too. I don't want there to be too much noise around the actual bug, but I should probably insert a "..." in there to make that clear. Hmm, and `extends Subject`, too. Will do. thanks

File: docs/bugpattern/ChainedAssertionLosesContext.md (hunk 4 above)
Comment (cpovirk) [reply docs/bugpattern/ChainedAssertionLosesContext.md]: Ditto.

**Pull request merged**